src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.R.cs

```diff
@@ -0,0 +1,223 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers.Text
+{
+    public static partial class Utf8Parser
+    {
+        //
+        // Parse an RFC1123 date string.
+        //
+        // 01234567890123456789012345678
+        // -----------------------------
+        // Tue, 03 Jan 2017 08:08:05 GMT
+        //
+        private static bool TryParseDateTimeOffsetR(ReadOnlySpan<byte> text, out DateTimeOffset dateTimeOffset, out int bytesConsumed)
+        {
+            if (text.Length < Utf8Constants.Rfc1123DateStringLength)
+            {
+                bytesConsumed = 0;
+                dateTimeOffset = default;
+                return false;
+            }
+
+            DayOfWeek dayOfWeek;
+            {
+                uint dow0 = text[0] & ~0x20u;
+                uint dow1 = text[1] & ~0x20u;
+                uint dow2 = text[2] & ~0x20u;
+                uint comma = text[3];
+                uint dowString = (dow0 << 24) | (dow1 << 16) | (dow2 << 8) | comma;
+                switch (dowString)
+                {
+                    case 0x53554E2c: dayOfWeek = DayOfWeek.Sunday; break;
```

Comment:
Might be worth adding a comment explaining what these magic numbers are.

Reply:
Added